core/unix/signal_linux.c

```diff
@@ -349,6 +349,13 @@ signalfd_thread_exit(dcontext_t *dcontext, thread_sig_info_t *info)
     TABLE_RWLOCK(sigfd_table, write, unlock);
 }
 
+bool
+is_repeat_handle_pre_extended_syscall_sigmasks(dcontext_t *dcontext)
+{
+    thread_sig_info_t *info = (thread_sig_info_t *)dcontext->signal_field;
+    return info->pre_syscall_app_sigprocmask_valid;
```

Comment:
We def couldn't handle a 'nested' epoll_pwait right now.

Reply:
But does the kernel handle that?  As mentioned I would not expect the kernel to bother with a stack of state: it would either stash the state it needs in the signal frame, or just say this isn't supported.

Reply:
I don't know, focus'sing on simpler things right now, e.g. pthread version doesn't work w/ DR. I can add a XXX comment mentioning this. I suggest to say we do not support "if" kernel does.

Reply:
Added XXX comment.